From 1cef8fe481de393c94e4709a28ea03d6dc9d7c2f Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Mon, 4 Mar 2013 17:19:20 +0800
Subject: [PATCH 130/231] ENGR00251209-4 msl-mx6: usb: Fix system hang when
 unload gadget module

At gadget module remove function, it closes the clock, but
at platform code, it still visits register. In fact,
The PHY has already been low power mode when driver's remove
before platform code is called.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 arch/arm/mach-mx6/usb_dr.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm/mach-mx6/usb_dr.c b/arch/arm/mach-mx6/usb_dr.c
index a8a21a0..6a70473 100644
--- a/arch/arm/mach-mx6/usb_dr.c
+++ b/arch/arm/mach-mx6/usb_dr.c
@@ -163,9 +163,6 @@ static void usbotg_uninit_ext(struct platform_device *pdev)
 {
 	otg_used--;
 	if (!otg_used) {
-		enter_phy_lowpower_suspend(pdev->dev.platform_data, true);
-		mdelay(3);
-
 		clk_disable(usb_phy1_clk);
 		clk_put(usb_phy1_clk);
 
-- 
1.7.10.4

